libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/for_loop.hpp

```diff
@@ -0,0 +1,1369 @@
+    ///
+    /// \returns  The \a for_loop_n_strided algorithm returns a
+    ///           \a hpx::future<void> if the execution policy is of
+    ///           type
+    ///           \a hpx::execution::sequenced_task_policy or
+    ///           \a hpx::execution::parallel_task_policy and returns \a void
+    ///           otherwise.
+    ///
+    template <typename ExPolicy, typename Rng, typename Size, typename... Args>
+    typename util::detail::algorithm_result<ExPolicy>::type for_loop_n_strided(
+        ExPolicy&& policy, Rng&& rng, Size size, Args&&... args);
+}}    // namespace hpx::ranges
+
+#else
+
+#include <hpx/config.hpp>
+#include <hpx/assert.hpp>
+#include <hpx/functional/tag_invoke.hpp>
+#include <hpx/modules/concepts.hpp>
+#include <hpx/modules/executors.hpp>
+#include <hpx/modules/iterator_support.hpp>
+#include <hpx/modules/type_support.hpp>
+#include <hpx/parallel/algorithms/for_loop.hpp>
+#include <hpx/parallel/util/detail/algorithm_result.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <iterator>
+#include <type_traits>
+#include <utility>
+
+namespace hpx { namespace ranges {
+    HPX_INLINE_CONSTEXPR_VARIABLE struct for_loop_t final
+      : hpx::functional::tag<for_loop_t>
+    {
+    private:
+        // clang-format off
+        template <typename ExPolicy, typename Iter, typename Sent, typename... Args,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_iterator<Iter>::value &&
+                hpx::traits::is_sentinel_for<Sent, Iter>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy>::type
+        tag_invoke(hpx::ranges::for_loop_t, ExPolicy&& policy, Iter first,
+            Sent last, Args&&... args)
+        {
+            static_assert(sizeof...(Args) >= 1,
+                "for_loop must be called with at least a function object");
+
+            using hpx::util::make_index_pack;
+            return parallel::v2::detail::for_loop(
+                std::forward<ExPolicy>(policy), first, last, 1,
+                typename make_index_pack<sizeof...(Args) - 1>::type(),
+                std::forward<Args>(args)...);
+        }
+
+        // clang-format off
+        template <typename Iter, typename Sent, typename... Args,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<Iter>::value &&
+                hpx::traits::is_sentinel_for<Sent, Iter>::value
+            )>
+        // clang-format on
+        friend void tag_invoke(
+            hpx::ranges::for_loop_t, Iter first, Sent last, Args&&... args)
+        {
+            static_assert(sizeof...(Args) >= 1,
+                "for_loop must be called with at least a function object");
+
+            return for_loop(parallel::execution::seq, first, last,
+                std::forward<Args>(args)...);
+        }
```

Comment:
I usually rather copy the code from the execution policy version instead of calling it to avoid to double tag-dispatch, but that's a minor issue.

Reply:
Good point. I can update them to not do that (including the non-ranges versions).

Reply:
I updated them to avoid going through the tag dispatch twice.